go/libraries/doltcore/sqle/diff_table.go

```diff
@@ -0,0 +1,324 @@
+
+	if err != nil {
+		return nil, err
+	}
+
+	toData, toSch, err := tableData(ctx, dt.toRoot, dt.name, dt.dEnv.DoltDB)
+
+	if err != nil {
+		return nil, err
+	}
+
+	fromConv, err := rowconv.RowConvForSuperSchema(fromSch, dt.superSch)
+
+	if err != nil {
+		return nil, err
+	}
+
+	toConv, err := rowconv.RowConvForSuperSchema(toSch, dt.superSch)
+
+	if err != nil {
+		return nil, err
+	}
+
+	sch := dt.joiner.GetSchema()
+	toCol, ok := sch.GetAllCols().GetByName(toCommit)
+
+	if !ok {
+		panic("missing required column")
+	}
+
+	fromCol, ok := sch.GetAllCols().GetByName(fromCommit)
+
+	if !ok {
+		panic("missing required column")
+	}
+
+	return NewDiffRowItr(dt.joiner, fromData, toData, fromConv, toConv, dt.fromCommitVal, dt.toCommitVal, fromCol.Tag, toCol.Tag), nil
+}
+
+var _ sql.RowIter = (*DiffRowItr)(nil)
+
+type DiffRowItr struct {
+	ad      *diff.AsyncDiffer
+	diffSrc *diff.RowDiffSource
+	sch     schema.Schema
+	to      string
+	from    string
+	fromTag uint64
+	toTag   uint64
+}
+
+func NewDiffRowItr(joiner *rowconv.Joiner, rowDataFrom, rowDataTo types.Map, convFrom, convTo *rowconv.RowConverter, from, to string, fromTag, toTag uint64) *DiffRowItr {
+	ad := diff.NewAsyncDiffer(1024)
+	ad.Start(context.TODO(), rowDataTo, rowDataFrom)
+
+	src := diff.NewRowDiffSource(ad, joiner)
+	src.AddInputRowConversion(convFrom, convTo)
+
+	return &DiffRowItr{ad, src, joiner.GetSchema(), to, from, fromTag, toTag}
+}
+
+func (itr *DiffRowItr) Next() (sql.Row, error) {
+	r, _, err := itr.diffSrc.NextDiff()
+
+	if err != nil {
+		return nil, err
+	}
+
+	r, err = r.SetColVal(itr.toTag, types.String(itr.to), itr.sch)
+
+	if err != nil {
+		return nil, err
+	}
+
+	r, err = r.SetColVal(itr.fromTag, types.String(itr.from), itr.sch)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return doltRowToSqlRow(r, itr.sch)
+}
+
+func (itr *DiffRowItr) Close() (err error) {
+	defer itr.ad.Close()
+	defer func() {
+		closeErr := itr.diffSrc.Close()
+
+		if err == nil {
+			err = closeErr
+		}
+	}()
+
+	return nil
+}
+
+func (dt *DiffTable) HandledFilters(filters []sql.Expression) []sql.Expression {
+	handled := make([]sql.Expression, 0, len(filters))
+	for _, f := range filters {
+		expression.Inspect(f, func(e sql.Expression) bool {
+			if e, ok := e.(*expression.GetField); ok {
+				if e.Table() == dt.Name() && e.Name() == toCommit || e.Name() == fromCommit {
+					handled = append(handled, f)
+					return false
+				}
+			}
+			return true
+		})
+	}
+
+	return handled
+}
+
+func (dt *DiffTable) WithFilters(filters []sql.Expression) sql.Table {
+	ctx := context.TODO()
+
+	for _, f := range filters {
+		var fieldName string
+		var value string
+		expression.Inspect(f, func(e sql.Expression) bool {
+			if e == nil {
+				return true
+			}
+
+			switch val := e.(type) {
+			case *expression.GetField:
+				fieldName = val.Name()
+			case *expression.Literal:
+				value = val.String()
+			}
+
+			return true
+		})
+
+		value = strings.Trim(value, " \t\n\r\"")
```

Comment:
Rather than panicking down below, you should either a) defer this verification logic to when you can return an error (PartitionRows), or b) save such as error to the dt struct and return it in PartitionRows